tasks/dotnettracer.py

```diff
@@ -0,0 +1,146 @@
+"""
+customaction namespaced tasks
+"""
+from __future__ import print_function
+
+import glob
+import os
+import shutil
+import sys
+
+from invoke import task
+from invoke.exceptions import Exit, ParseError, Failure, UnexpectedExit
+from .ssm import get_signing_cert, get_pfx_pass
+
+projects_to_restore = [
+    "Datadog.Trace", "Datadog.Trace.AspNet", "Datadog.Trace.ClrProfiler.Managed", "Datadog.Trace.ClrProfiler.Managed.Core", "Datadog.Trace.ClrProfiler.Managed.Loader"
+]
+@task
+def build(ctx, vstudio_root=None, arch="x64", major_version='7', debug=False):
+    """
+    Build the custom action library for the agent
+    """
+
+    if sys.platform != 'win32':
+        print("DotNet Tracer build is only for Win32")
+        raise Exit(code=1)
+
+    print("arch is {}".format(arch))
+    this_dir = os.getcwd()
+    solution_dir = os.getcwd()
+    tracer_home = os.path.join(solution_dir, "tracer_home")
+    msi_home = os.path.join(solution_dir, "msi")
+    
+    pfxfile = None
+    pfxpass = None
+    remove_pfx = True
+    cmd = ""
+    configuration = "Release"
+    if debug:
+        configuration = "Debug"
+    ctx.run("nuget restore {solution_dir}\\Datadog.Trace.Minimal.sln".format(solution_dir=solution_dir))
+    for proj in projects_to_restore:
+        try:
+            # some projects may not exist in older commits, which is fine
+            nuget_cmd = "nuget restore {solution_dir}\\src\\{proj}\\{proj}.csproj -PackagesDirectory packages".format(
+                solution_dir = solution_dir,
+                proj=proj
+            )
+            ctx.run(nuget_cmd)
+        except:
+            print("Failed to restore nuget package {proj}\n".format(proj=proj))
+            pass
+
+    cmd = "msbuild {solution_dir}\\Datadog.Trace.proj /t:{target} /p:Configuration={config} /p:TracerHomeDirectory={tracer_home} /p:RunWixToolsOutOfProc=true /p:MsiOutputPath={msihome}"
```

Comment:
We should pass the `arch` parameter and change it to `All` 
```suggestion
    cmd = "msbuild {solution_dir}\\Datadog.Trace.proj /t:{target} /p:Platform={arch} /p:Configuration={config} /p:TracerHomeDirectory={tracer_home} /p:RunWixToolsOutOfProc=true /p:MsiOutputPath={msihome}"
```